### DISCUSSION TITLE
How to specify whether to link ASan statically or dynamically?

### DISCUSSION THREAD

**Discussion by AminoACID123** [zephyrproject-rtos/zephyr#65451] [Q&A]
I want to compile zephyr with address sanitizer enabled. I wonder if it is possible to choose between static linking and dynamic linking of ASan library.

**Comment by omkar3141** [comment]
How to [enable address sanitizer](https://docs.zephyrproject.org/latest/boards/native/native_sim/doc/index.html#address-sanitizer-asan) at compile time for Zephyr RTOS?
To do this, you can try passing CONFIG_ASAN=y. For example: 

`west build -b native_posix -- -DCONFIG_ASAN=y`.
